packages/sources/ion.au/src/transport/total_reserve.ts

```diff
@@ -0,0 +1,66 @@
+import { HttpTransport } from '@chainlink/external-adapter-framework/transports'
+import { BaseEndpointTypes } from '../endpoint/total_reserve'
+
+export interface ResponseSchema {
+  total_reserve: number
+  errorMessage?: string
+}
+
+export type HttpTransportTypes = BaseEndpointTypes & {
+  Provider: {
+    RequestBody: never
+    ResponseBody: ResponseSchema
+  }
+}
+export const httpTransport = new HttpTransport<HttpTransportTypes>({
+  prepareRequests: (params, config) => {
+    return params.map((param) => {
+      return {
+        params: [param],
+        request: {
+          baseURL: config.API_ENDPOINT,
+          url: '/',
+          headers: {},
+          params: {},
+        },
+      }
+    })
+  },
+  parseResponse: (params, response) => {
+    if (response.data.errorMessage) {
+      return params.map((param) => {
+        return {
+          params: param,
+          response: {
+            errorMessage: `There was an error from the source API. ${response.data.errorMessage}`,
+            statusCode: 502,
+          },
+        }
+      })
```

Comment:
Given that there's no input parameters used in the endpoint, all of these returns can be simplified:
```suggestion
      return [{
          params,
          response: {
            errorMessage: `There was an error from the source API. ${response.data.errorMessage}`,
            statusCode: 502,
          },
        }]
```

Reply:
More info on HTTP transports [here](https://github.com/smartcontractkit/ea-framework-js/blob/main/docs/components/transport-types/http-transport.md) if you're curious

Reply:
I made this change locally and when I tested it I found that the adapter reverted to the following error every request. Is this necessary?

504 - `The EA has not received any values from the Data Provider for the requested data yet. Retry after a short delay, and if the problem persists raise this issue in the relevant channels.`

Reply:
My bad, I wrote it quickly and didn't spend a lot of time checking that it matched. I tested this locally and it should work:
```typescript
export const httpTransport = new HttpTransport<HttpTransportTypes>({
  prepareRequests: (params, config) => {
    return {
      params,
      request: {
        baseURL: config.API_ENDPOINT,
        url: '/',
        headers: {},
        params: {},
      },
    }
  },
  parseResponse: (params, response) => {
    if (response.data.errorMessage) {
      return [
        {
          params: {},
          response: {
            errorMessage: `There was an error from the source API. ${response.data.errorMessage}`,
            statusCode: 502,
          },
        },
      ]
    }
    if (!response.data.total_reserve && response.data.total_reserve !== 0) {
      return [
        {
          params: {},
          response: {
            errorMessage: `The data provider didn't return any value for total_reserve`,
            statusCode: 502,
          },
        },
      ]
    }

    const result = response.data.total_reserve
    return [
      {
        params: {},
        response: {
          result,
          data: {
            result,
          },
        },
      },
    ]
  },
})
```